web-client/public/js/upload.js

```diff
@@ -337,8 +437,9 @@ export const upload = function () {
         ] : [
             (grnState.workbook.meta2 !== undefined && "optimization_diagnostics")
         ];
-        additionalsheets = additionalsheets.filter(x => x !== false && x !== "optimization_parameters");
-        additionalsheets = ["optimization_parameters", ...additionalsheets].sort();
+        additionalsheets = additionalsheets.filter(x => (x !== false && -1 !== optionalAdditionalSheets.indexOf(x)));
```

Comment:
`x !== false` is more verbose than it has to be, yes?

Reply:
…unless there is a specific intent here to also allow `x` values of `undefined` and `null` to make this condition true. And if so, that is somewhat irregular so therefore worth a comment

Reply:
I changed the line to be: ```additionalsheets = additionalsheets.filter(sheet => (sheet && -1 !== optionalAdditionalSheets.indexOf(sheet)));``` to remove the verbose statement and add more clarity, changing `x` to `sheet`.